mantidproject/mantid
# Project Recovery: Use a counter to save workspaces instead of name

**Description of work.**
Replaced workspace names when saving with a counter for the number of workspaces that have already been saved

**To test:**
- Load a workspace
- Wait for workspace to be saved
- Check .mantid/recovery/computername/mostrecentcheckpoint/ and see that there should be a 0.py instead of the workspace name
- Retry with multiple workspaces



Fixes #23258  


#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
